Foundation/NSFileManager.swift

```diff
@@ -417,7 +417,38 @@ public class NSFileManager : NSObject {
     }
     
     public func copyItemAtPath(srcPath: String, toPath dstPath: String) throws {
-        NSUnimplemented()
+        let fd_from = open(srcPath, O_RDONLY)
+        if fd_from < 0 {
+            throw _NSErrorWithErrno(errno, reading: true, path: dstPath, url: nil, extraUserInfo: nil)
+        }
+        defer { precondition(close(fd_from) >= 0) }
+        let permission = (try! attributesOfItemAtPath(srcPath)[NSFilePosixPermissions] as! NSNumber).unsignedShortValue
+        let fd_to = open(dstPath, O_WRONLY | O_CREAT | O_EXCL, permission)
+        if fd_to < 0 {
+            throw _NSErrorWithErrno(errno, reading: false, path: dstPath, url: nil, extraUserInfo: nil)
+        }
+        defer { precondition(close(fd_to) >= 0) }
+        
+        var buf = [UInt8](count: 4096, repeatedValue: 0)
+        
+        while true {
+            let nread = read(fd_from, &buf, buf.count)
```

Comment:
It should be; perhaps the import isn't in the Glibc overlay for linux for #include <sys/sendfile.h>
It could be re-exposed like _CFOpenFileWithMode has to be re-exported to swift.

it would be totally reasonable to make a platform abstraction for copyfile in CFPlatform.c since Darwin is different than linux and if ever there is a Windows port there is a different function there too.